src/DIRAC/FrameworkSystem/DB/ProxyDB.py

```diff
@@ -395,6 +398,9 @@ def deleteProxy(self, userDN, userGroup=None, proxyProvider=None):
             return S_ERROR(", ".join(errMsgs))
         return result
 
+    _cache_getProxyStrength = TTLCache(maxsize=1000, ttl=600)
+
+    @cached(_cache_getProxyStrength, lock=Lock())
```

Comment:
I am not sure it is a good idea to have a lock there. Given that the result is deterministic, we don't really care if the cache is filled several times, do we ?

Reply:
I am not sure, but according to the documentation these classes are not thread-safe, so I'd leave the lock. At the same time I think to put `(maxsize=100, ttl=3600)`.